numpy/core/src/umath/ufunc_object.c

```diff
@@ -2216,26 +2244,50 @@ PyUFunc_GeneralizedFunction(PyUFuncObject *ufunc,
     /* Get all the arguments */
     retval = get_ufunc_arguments(ufunc, args, kwds,
                 op, &order, &casting, &extobj,
-                &type_tup, &subok, NULL, &axes);
+                &type_tup, &subok, NULL, &axes, &axis, &keepdims);
     if (retval < 0) {
-        goto fail;
+        return retval;
+    }
+    /*
+     * If keepdims was passed in (and thus changed from the initial value
+     * on top), check the gufunc is suitable, i.e., that:
+     * - its input arguments share a single core dimension;
```

Comment:
This is actually not strictly necessary (and checked); signatures like `(i),(j)->()` are fine for `keepdims`. Will change after having had further review.